FirebaseAuth/Sources/Swift/Auth/Auth.swift

```diff
@@ -1644,27 +1643,28 @@ extension Auth: AuthInterop {
                                                     appCheck: appCheck)
     self.backend = backend
     self.authDispatcher = authDispatcher
+
+    let keychainServiceName = Auth.keychainServiceName(for: app)
+    keychainServices = AuthKeychainServices(service: keychainServiceName,
+                                            storage: keychainStorageProvider)
+    storedUserManager = AuthStoredUserManager(
+      serviceName: keychainServiceName,
+      keychainServices: keychainServices
+    )
+
     super.init()
     requestConfiguration.auth = self
 
-    protectedDataInitialization(keychainStorageProvider)
+    protectedDataInitialization()
   }
 
-  private func protectedDataInitialization(_ keychainStorageProvider: AuthKeychainStorage) {
+  private func protectedDataInitialization() {
     // Continue with the rest of initialization in the work thread.
     kAuthGlobalWorkQueue.async { [weak self] in
       // Load current user from Keychain.
       guard let self else {
         return
       }
-      if let keychainServiceName = Auth.keychainServiceName(forAppName: self.firebaseAppName) {
-        self.keychainServices = AuthKeychainServices(service: keychainServiceName,
-                                                     storage: keychainStorageProvider)
-        self.storedUserManager = AuthStoredUserManager(
-          serviceName: keychainServiceName,
-          keychainServices: self.keychainServices
-        )
-      }
```

Comment:
It seems safer to move this into the initializer since we know all this at initialization time. Delaying it forces us to add bangs! to property definitions since they are initialized outside of the initializer (scroll down to last few diffs).